src/common/structures/marker_status.py

```diff
@@ -0,0 +1,13 @@
+from typing import Final
+from enum import IntEnum
+
+
+class MarkerStatus:
+
+    class Marker(IntEnum):
+        STOPPED: Final[int] = 0
+        RUNNING: Final[int] = 1
+        FAILURE: Final[int] = 2
+
+    status: Marker
```

Comment:
No initial value, I suggest to have an initializer for each of status (STOPPED) and errors (empty list)